lineapy/plugins/taskgen.py

```diff
@@ -0,0 +1,221 @@
+"""
+This taskgen files contains helper functions to create tasks.
+"""
+from typing import Dict
+
+from lineapy.graph_reader.artifact_collection import ArtifactCollection
+from lineapy.graph_reader.node_collection import ArtifactNodeCollection
+from lineapy.plugins.session_writers import BaseSessionWriter
+from lineapy.plugins.task import DagTaskBreakdown, TaskDefinition
+from lineapy.plugins.utils import load_plugin_template
+
+
+def get_task_definitions(
+    artifact_collection: ArtifactCollection,
+    pipeline_name: str,
+    task_breakdown: DagTaskBreakdown,
+) -> Dict[str, TaskDefinition]:
+    """
+    get_task_definitions returns a dictionary of TaskDefinitions
+
+    This function breaks down the artifact_collection into tasks based on the
+    task_breakdown parameter. This will give the main bulk of tasks that should
+    be included in a pipeline dag file.
+
+    Returns a `task_definitions` dictionary, which maps a key corresponding to the task name to
+    Linea's TaskDefinition object.
+    Specific framework implementations of PipelineWriters should serialize the TaskDefinition
+    objects to match the format for pipeline arguments that is expected by that framework.
+    """
+    if task_breakdown == DagTaskBreakdown.TaskAllSessions:
+        return get_allsessions_task_definition(pipeline_name)
+    elif task_breakdown == DagTaskBreakdown.TaskPerSession:
+        return get_session_task_definition(artifact_collection, pipeline_name)
+    elif task_breakdown == DagTaskBreakdown.TaskPerArtifact:
+        return get_artifact_task_definitions(
+            artifact_collection, pipeline_name
+        )
+    else:
+        raise ValueError(
+            f"Task breakdown granularity {task_breakdown} is not currently supported."
+        )
+
+
+def get_artifact_task_definitions(
+    artifact_collection: ArtifactCollection, pipeline_name: str
+) -> Dict[str, TaskDefinition]:
+    """
+    get_artifact_task_definitions returns a task definition for each artifact the pipeline produces.
+    This may include tasks that produce common variables that were not initially defined as artifacts.
+    """
+    task_definitions: Dict[str, TaskDefinition] = dict()
```

Comment:
My understanding is that item order does matter for `task_definition`. To better signify this (to another engineer/contributor), we may want to use `OrderedDict` instead of `Dict` (although the latter keeps items in order too).

Reply:
I actually think for now just a dict() is fine. The order should be dictated by a TaskGraph, which right now is not attached to these functions but eventually should be returned here as well.

Reply:
Yeah, realized this comment is outdated as per my later comments.